java-client/src/test/java/org/opensearch/client/opensearch/integTest/AbstractAggregationRequestIT.java

```diff
@@ -190,13 +189,12 @@ private String getErrorMetadata(ErrorCause errorCause) {
 	public static class ProductDetails {
 		private String name;
 		private int cost;
-		@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS")
-		private Date expDate;
+		private String expDate;
```

Comment:
This change is not very straightforward: definitely :+1: to move off the `Date`, but using string to pass the dates through is not a best option I think:
 - should we use `LocateDateTime` / `Instant` instead` ?
 - should be use the timestamp value (long)?

Reply:
String allows to insert date in ISO format without any transformations.

LocateDateTime and Instant are requiring:
```
@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
private Instant expDate;
```
build.gradle.kts:
`testImplementation("com.fasterxml.jackson.datatype", "jackson-datatype-jsr310", jacksonVersion)`

JacksonJsonpMapper -> findAndRegisterModules method used:
```
new ObjectMapper()
                .findAndRegisterModules()
                .configure(SerializationFeature.INDENT_OUTPUT, false)
                .setSerializationInclusion(JsonInclude.Include.NON_NULL)
```

private Long expDate; does not require anything more.
Seems it works but the minus is that the Document is created with numeric values:
```{
  "_index": "test-date-range-aggregation",
  "_type": "_doc",
  "_id": "1",
  "_score": 1,
  "_source": {
    "name": "egg",
    "cost": 2,
    "expDate": 1676937600000
  }
}
```

So I'm not sure what is better than simple String ;P

Reply:
> String allows to insert date in ISO format without any transformations.

That is what `@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = ISO) is for, piratically this is how the feature should be used (yes, it simplifies tests).

Reply:
Ok, let's stick with Long expDate.